lib/utils.js

```diff
@@ -44,54 +54,48 @@ function mergeObject(target, source) {
     const sourceKeys = Object.keys(source);
     for (let i = 0; i < sourceKeys.length; ++i) {
         const key = sourceKeys[i];
-        if (source[key] === undefined) {
+
+        // If shouldRemoveNullObjectValues is true, we want to remove null values from the merged object
+        if (shouldRemoveNullObjectValues && _.isNull(source[key])) {
```

Comment:
Can we remove lodash from this function? Lodash is slow and this function is used everywhere.

Reply:
I think performance is a good concern, I also agree to remove underscore and lodash for parts like this. I've also found those libraries lead to performance problems in low-lying libs like this.